syntaxes/sourcepawn.tmLanguage.json

```diff
@@ -27,14 +27,20 @@
 			}
 		},
 		{
-			"match": "(\\#define)\\s([^ ]*)",
+			"match": "(\\#define)\\s*([A-Z0-9_]*)\\s*((\"[.]*\")|(-?[0-9.]*))",
```

Comment:
The dot in a character set only matches the character `.`, `[.]*` should just be `.*` in the group for strings.